### ISSUE TITLE
[OpenBMC] lots of redfined functions for openbmc - error shows up with bypass

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#4856]
@zet809 @xuweibj Are you aware of this? 

```
[root@briggs01 ~]# XCATBYPASS=1 rpower mid05tor12cn05 state
Subroutine unsupported redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 102.
Subroutine handled_commands redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 120.
Subroutine preprocess_request redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 721.
Subroutine retry_after redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 846.
Subroutine retry_check_times redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 862.
Subroutine process_request redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 904.
Subroutine parse_args redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1129.
Subroutine parse_command_status redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1431.
Subroutine fork_process_login redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2130.
Subroutine get_functional_software_ids redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2161.
Subroutine parse_node_info redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2188.
Subroutine gen_send_request redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2269.
Subroutine deal_with_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2342.
Subroutine process_debug_info redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2472.
Subroutine login_request redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2495.
Subroutine login_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2531.
Subroutine rpower_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2557.
Subroutine rinv_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2754.
Subroutine getopenbmccons redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2914.
Subroutine rsetboot_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 2945.
Subroutine rbeacon_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3004.
Subroutine reventlog_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3043.
Subroutine is_callout_event_data redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3158.
Subroutine parse_event_data redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3184.
Subroutine rspconfig_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3263.
Subroutine rspconfig_api_config_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3590.
Subroutine rspconfig_sshcfg_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3696.
Subroutine sshcfg_process redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3735.
Subroutine rspconfig_dump_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3785.
Subroutine dump_download_process redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 3920.
Subroutine dump_download_all_process redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4003.
Subroutine rvitals_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4025.
Subroutine rflash_response redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4158.
Subroutine rflash_upload redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4500.
Subroutine is_valid_config_api redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4620.
Subroutine build_config_api_usage redefined at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 4652.
Tue Feb 27 11:22:56 2018 OpenBMC: [openbmc_debug]
Tue Feb 27 11:22:57 2018 mid05tor12cn05: [openbmc_debug] curl -k -c cjar -H "Content-Type: application/json" -d '{ "data": ["root", "xxxxxx"] }' https://172.11.139.5/login
Tue Feb 27 11:22:58 2018 mid05tor12cn05: [openbmc_debug] login_response 200 OK
Tue Feb 27 11:22:58 2018 mid05tor12cn05: [openbmc_debug] curl -k -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.5/xyz/openbmc_project/state/enumerate
Tue Feb 27 11:22:58 2018 mid05tor12cn05: [openbmc_debug] rpower_status_response 200 OK
mid05tor12cn05: on
```

Seems to have to do with openbmc.pm and openbmc2.pm..... can we fix this?

**Comment by immarvin** [comment]
`xcatd -f` also complains these messages

**Comment by daniceexi** [comment]
@hu-weihua could we use CI syntax check to filter out these kinds of warning information?

**Comment by xuweibj** [comment]
We know this, this is because that we call "parse_args" of openbmc.pm for rspconfig args check in openbmc2.pm. Have talked with @zet809 , I think we should rewrite it in openbmc2.pm.

**Comment by zet809** [comment]
Yes, @whowutwut , the message is introduced when try to support rspconfig in python, I reuse the openbmc::parse_args function. If not better method, I will rewrite parse_args in openbmc2.pm.
Thx, @whowutwut

**Comment by zet809** [comment]
Refined in PR #4944 and have been included in latest 2.14 snapshot build, so @whowutwut , shall I close this issue?
```
[root@c910f03c11k08 xcat]# lsxcatd -v
Version 2.14 (git commit 7bae503907bc8fb341662f6974319a1c791cf42b, built Mon Mar 19 06:15:48 EDT 2018)
[root@c910f03c11k08 xcat]# XCATBYPASS=1 rpower f6u03 state
Mon Mar 19 22:35:33 2018 f6u03: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://10.6.3.100/login -d '{"data": ["root", "xxxxxx"]}'
Mon Mar 19 22:35:33 2018 f6u03: [openbmc_debug] login 200 OK
Mon Mar 19 22:35:33 2018 f6u03: [openbmc_debug] list_power_states curl -k -c cjar -b cjar -X GET -H "Content-Type: application/json" https://10.6.3.100/xyz/openbmc_project/state/enumerate
Mon Mar 19 22:35:33 2018 f6u03: [openbmc_debug] list_power_states 200 OK
f6u03: off
```

**Comment by zet809** [comment]
Close this issue based on the verification result above.
